### PR TITLE
install: restrict access permissions on /boot/ignition{,/config.ign}

### PR DESCRIPTION
**Pull request by bgilbert**
The Ignition config may contain secrets.  Don't expose it, or anything in its parent directory, to unprivileged processes.

### CODE REVIEW

File: src/install.rs [hunk 1]
```diff
@@ -262,6 +264,13 @@ fn write_ignition(
                 config_dest.display()
             )
         })?;
+    // Ignition config may contain secrets; restrict to root
+    set_permissions(&config_dest, Permissions::from_mode(0o600)).with_context(|| {
```
Comment (travier) [inline src/install.rs:268]: I think we should set the whole folder to `700` (in the `create_dir_all` above)

**Comment by bgilbert** [comment]
Updated.

**Review by travier — APPROVED** [review]

**Comment by travier** [comment]
This _might_ benefit from a conversion to https://github.com/coreos/openat-ext but this can definitely be done later.

**Review by jlebon — APPROVED** [review]

**Comment by jlebon** [comment]
Probably worth mentioning https://github.com/coreos/fedora-coreos-tracker/issues/889 in the commit message for posterity.

On the other side of this, we should consider deleting the file after it's used. This is discussed in the tracker issue.

**Pull request merged**